tools/python/paladin_claiming/recycle_bribes.py

```diff
@@ -0,0 +1,195 @@
+                claim["proofs"],
+            )
+
+            token_contract = w3.eth.contract(
+                address=Web3.to_checksum_address(claim["token"]), abi=erc20_abi
+            )
+            decimals = token_contract.functions.decimals().call()
+            amount = int(claim["amount"]) / (10**decimals)
+
+            f.write(
+                f"{chain_name},{claim['token']},{claim['gauge']},{amount},{claim['amount']}\n"
+            )
+
+    return csv_path
+
+
+def deposit_hh_bribes(csv_path: str, chain_name: str, builder: SafeTxBuilder):
+    """
+    - generate tx to deposit 70% of claimed bribe amounts on HH
+    - save remaining 30% (DAO fee) to separate csv
+    - output final transaction payload for claims and deposits
+    """
+    chain_addrs = CHAIN_ADDRS[chain_name]
+
+    chain_dir = get_chain_dirs(CLAIM_OUTPUT_DIR, chain_name)
+    sell_csv_path = (
+        chain_dir / f"paladin_tokens_to_sell_{chain_name}_{CURRENT_DATE}.csv"
+    )
+    with open(sell_csv_path, "w") as f:
+        f.write("chain,token_address,amount,amount_mantissa\n")
+
+    # Aggregate claimed bribes by amounts by token and gauge
+    aggregated_bribes = {}
+    with open(csv_path, "r") as f:
+        next(f)
+        for line in f:
+            _, token, gauge, _, amount_mantissa = line.strip().split(",")
+            key = (token, gauge)
+            aggregated_bribes[key] = aggregated_bribes.get(key, 0) + int(
+                amount_mantissa
+            )
+
+    token_sell_amounts = {}
+
+    # Process aggregated bribes
+    for (token, gauge), amount_mantissa in aggregated_bribes.items():
+        print(
+            f"depositing bribe on HH - token: {token}, gauge: {gauge}, amount: {amount_mantissa}"
+        )
+
+        brib_token = SafeContract(token, erc20_abi)
+        briber = SafeContract(chain_addrs["bal_briber"], bribe_market_abi)
+
+        prop_hash = get_prop_hash(gauge)
+        assert prop_hash, f"no prop hash for {gauge}"
+
+        # deposit 70%
+        bribe_amount_mantissa = int(amount_mantissa * 0.7)
+
+        # sell 30% later
+        token_sell_amounts[token] = (
+            token_sell_amounts.get(token, 0) + amount_mantissa - bribe_amount_mantissa
+        )
+
+        brib_token.approve(chain_addrs["bribe_vault"], bribe_amount_mantissa)
+        briber.depositBribe(prop_hash, token, bribe_amount_mantissa, 0, 2)
+
+    # output sell amounts to csv
+    with open(sell_csv_path, "a") as f:
+        for token, sell_amount_mantissa in token_sell_amounts.items():
+            token_contract = w3_by_chain[chain_name].eth.contract(
+                address=Web3.to_checksum_address(token), abi=erc20_abi
+            )
+            decimals = token_contract.functions.decimals().call()
+            amount = sell_amount_mantissa / (10**decimals)
+            f.write(f"{chain_name},{token},{amount},{sell_amount_mantissa}\n")
+
+    payload_chain_dir = get_chain_dirs(PAYLOAD_DIR, chain_name)
+    builder.output_payload(
```

Comment:
can this use json dump with `indent=2` somehow?

Reply:
this would have to be done inside SafeTxBuilder. raised an issue here: https://github.com/BalancerMaxis/bal_tools/issues/54